src/transformers/models/bert/modeling_bert.py

```diff
@@ -1647,6 +1643,8 @@ def __init__(self, config):
         self.dropout = nn.Dropout(config.hidden_dropout_prob)
         self.classifier = nn.Linear(config.hidden_size, config.num_labels)
 
+        self.loss_fct = CrossEntropyLoss()
```

Comment:
If we are switching to `F.cross_entropy` why this addition was needed?

Reply:
I prefered creating an instance of CrossEntropyLoss because the ignore_index attribute is being used later in the forward pass:

```
 active_labels = torch.where(                    
     active_loss, labels.view(-1), torch.tensor(loss_fct.ignore_index).type_as(labels)
 )
```

Here, the default value is being used (-100) but I thought it would be clearer that way than hardcoding the ignore_index value.

Reply:
That makes sense!

So if the class approach works, why did this PR replace a lot of class `CrossEntropyLoss` with `F.cross_entropy`?

Reply:
Because the class are created in the forward pass, making tracing impossible (the tracer complains about it).

Reply:
Aha! I missed this important tidbit - thank you for making it clear to me - this makes total sense then.

Functional API seems to be the way to go then, since as you're saying it's not always used!

All is good here!